universe/base.go

```diff
@@ -245,6 +258,72 @@ func (a *MintingArchive) RegisterIssuance(ctx context.Context, id Identifier,
 	return issuanceProof, nil
 }
 
+func (a *MintingArchive) getPrevAssetSnapshot(ctx context.Context,
+	uniID Identifier, newProof proof.Proof) (*proof.AssetSnapshot, error) {
+
+	// If this is a genesis proof, then there is no previous asset (and
+	// therefore no previous asset snapshot).
+	if newProof.Asset.HasGenesisWitness() {
+		return nil, nil
+	}
+
+	// Query for proof associated with the previous asset.
+	prevID := newProof.Asset.PrevWitnesses[0].PrevID
+	prevOutPoint := prevID.OutPoint
+
+	var allZeros [32]byte
+	if reflect.DeepEqual(prevOutPoint.Hash.CloneBytes(), allZeros[:]) {
```

Comment:
Can just compare `if prevOutPoint.Hash == allZeroes {`. Though I think this shouldn't be necessary, as that should be covered in the `HasGenesisWitness()`. Let's discuss offline.

Reply:
Okay, I know why this is the case. If this is a split commitment, you instead need to use the root asset's `PrevID`. See this commit from the burn PR: https://github.com/lightninglabs/taproot-assets/blob/e1af71979112702614f8d1e72aae914cab9cd270/asset/witness.go#L19

Reply:
Maybe we could generalize that part into a `func GetFirstPrevOut(witness TxWitness)` function?

Reply:
I'll definitely spin this bit of logic into a function. I think it will help document (in a single spot) what's going on.